# HG changeset patch
# User David Bateman <dbateman@free.fr>
# Date 1238790221 -7200
#      Fri Apr 03 22:23:41 2009 +0200
# Node ID 136e72b9afa8ce589acf8f7babfd2b8212e5e96b
# Parent  b7210faa3ed0d9b52d730309c88c46668d1cb836
correct indexing of cdata for scatter

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2008-04-03  David Bateman  <dbateman@free.fr>
+
+	* plot/__scatter__.m: correct indexing of cdata.x
+
 2009-04-03  Jaroslav Hajek  <highegg@gmail.com>
 
 	* optimization/fminunc.m: New function.
 	* optimization/Makefile.in: Update.
 	* optimization/__dogleg__: Allow general quadratics.
 
 2009-04-02  Ben Abbott <bpabbott@mac.com>
 
diff --git a/scripts/plot/__scatter__.m b/scripts/plot/__scatter__.m
--- a/scripts/plot/__scatter__.m
+++ b/scripts/plot/__scatter__.m
@@ -144,17 +144,18 @@ function hg = __scatter__ (varargin)
 	set(h, "markerfacecolor", c); 
       endif
     endfor
   else
     for i = 1 : numel (x)
       h = __go_patch__ (hg, "xdata", x(i), "ydata", y(i), "zdata", z(i,:),
 			"faces", 1, "vertices", [x(i), y(i), z(i,:)], 
 			"facecolor", "none", "edgecolor", "flat", 
-			"cdata", c(i), "marker", marker, "markersize", s(i), 
+			"cdata", reshape(c(i,:),[1,size(c)(2:end)]), 
+			"marker", marker, "markersize", s(i), 
 			"linestyle", "none");
       if (filled)
 	set(h, "markerfacecolor", "flat"); 
       endif
     endfor
     ax = get (hg, "parent");
     clim = get (ax, "clim");
     if (min(c(:)) < clim(1))
@@ -210,19 +211,33 @@ function update_data (h, d)
     c1 = repmat (c1, numel (x1), 1);
   endif
   size1 = get (h, "sizedata");
   if (numel (size1) == 1)
     size1 = repmat (size1, numel (x1), 1);
   endif
   hlist = get (h, "children");
   if (ischar (c1))
-    for i = 1 : length (hlist)
-      set (hlist(i), "vertices", [x1(i), y1(i), y2(i)], "cdata", c1,
-	   "markersize", size1(i));
-    endfor
+    if (isempty (z1))
+      for i = 1 : length (hlist)
+	set (hlist(i), "vertices", [x1(i), y1(i)], "cdata", c1,
+	     "markersize", size1(i));
+      endfor
+    else
+      for i = 1 : length (hlist)
+	set (hlist(i), "vertices", [x1(i), y1(i), z1(i)], "cdata", c1,
+	     "markersize", size1(i));
+      endfor
+    endif
   else
-    for i = 1 : length (hlist)
-      set (hlist(i), "vertices", [x1(i), y1(i), y2(i)], "cdata", c1(i,:),
-	   "markersize", size1(i));
-    endfor
+    if (isempty (z1))
+      for i = 1 : length (hlist)
+	set (hlist(i), "vertices", [x1(i), y1(i)], "cdata", 
+	     reshape(c1(i,:),[1, size(c1)(2:end)]), "markersize", size1(i));
+      endfor
+    else
+      for i = 1 : length (hlist)
+	set (hlist(i), "vertices", [x1(i), y1(i), z1(i)], "cdata", 
+	     reshape(c1(i,:),[1, size(c1)(2:end)]), "markersize", size1(i));
+      endfor
+    endif
   endif
 endfunction
